airflow/cli/commands/connection_command.py

```diff
@@ -195,6 +212,16 @@ def connections_add(args):
     if has_json and has_uri:
         raise SystemExit('Cannot supply both conn-uri and conn-json')
 
+    if has_type:
+        is_valid_conn_type = _valid_conn_type(args.conn_type)
+        if not is_valid_conn_type:
+            warnings.warn(f'The type provided to --conn-type is invalid: {args.conn_type}')
+            warnings.warn(
+                f'Supported --conn-types are:{_get_connection_types()}.'
+                'Hence overriding the conn-type with generic'
+            )
```

Comment:
Not sure `warnings.warn` is the right thing to do (not faimilar with how commands log things). Maybe `log.warning` is better?